### PR TITLE
github action for build and test

### LINKED ISSUE #101: CI Pipeline
Set up a CI(Continuous Integration) pipeline for the doc generator service using github actions.
This pipeline will automate the build, testing.

### PR DESCRIPTION
**Pull request by Tushar98644**
fixes #101 

@radhay-1199 Kept the pipeline as default as it can be , awaiting your review for the specifications that need to be added

### CODE REVIEW

**Comment by AnshulMalik** [comment]
Thanks for the PR @Tushar98644 . 
Just a couple of things, can we update the branches to be `Roadmap-2023`? And we are using `yarn` instead of `npm` in this project, so that would also be preferred.

**Comment by Tushar98644** [comment]
> Thanks for the PR @Tushar98644 . Just a couple of things, can we update the branches to be `Roadmap-2023`? And we are using `yarn` instead of `npm` in this project, so that would also be preferred.

I have updated it from npm to yarn. For update the branch to Roadmap-2023, do you mean changing the source branch to Roadmap-2023?

**Comment by AnshulMalik** [comment]
> For update the branch to Roadmap-2023, do you mean changing the source branch to Roadmap-2023?

Yes, I meant the source branch, which is currently `main` in the action file.

**Comment by Tushar98644** [comment]
> > For update the branch to Roadmap-2023, do you mean changing the source branch to Roadmap-2023?
> 
> Yes, I meant the source branch, which is currently `main` in the action file.

Currently github does not allow the source branch to be changed after a pr is created , should I create a new pr?

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Build and Test
+
+on:
+  push:
+    branches:
+      - main
```
Comment (AnshulMalik) [inline .github/workflows/build-test.yml]: this branch i was talking about.

Comment (Tushar98644) [reply .github/workflows/build-test.yml]: updated

**Review by AnshulMalik — APPROVED** [review]

**Comment by AnshulMalik** [comment]
@Tushar98644 this step is failing `cp nest-cli.json dist/nest-cli.json`

**Comment by Tushar98644** [comment]
> @Tushar98644 this step is failing `cp nest-cli.json dist/nest-cli.json`

I have removed that , that task was necessary if a build/dist was present , but there is no such folder

**Comment by AnshulMalik** [comment]
Lockfile needs to be updated @Tushar98644

**Comment by Tushar98644** [comment]
> Lockfile needs to be updated @Tushar98644

updated , now it works fine

**Pull request merged**